### PR TITLE
Client Supplemental data bugfix

### PR DESCRIPTION
**Pull request by ttoomey**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch
- use a merge-commit or rebase strategy for PRs targeting the stable branch

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)

Looks like the previous bugfix wasn't correctly applied. Fix and add tests to prove it. We were still seeing the behavior of too many source clients shown on a the client data source

## Type of change
Bug fix

## testing
visit [QA client page](https://qa-warehouse.openpath.host/hmis_supplemental/data_sets/1/client_data_sets/19574). Should only one set of values for a given client in the data source 

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] I have updated the documentation (or not applicable)
- [x] I have added spec tests (or not applicable)
- [x] I have provided testing instructions in this PR or the related issue (or not applicable)

### CODE REVIEW

File: drivers/hmis_supplemental/app/controllers/hmis_supplemental/client_data_sets_controller.rb [hunk 1]
```diff
@@ -39,9 +39,10 @@ def show
     protected
 
     def source_clients
+      # order is important here, source_visible_to appears to clobber the data source condition
       @client.source_clients.
-        where(data_source_id: @data_set.data_source_id).
         source_visible_to(current_user).
+        where(data_source_id: @data_set.data_source_id).
```
Comment (ttoomey) [inline drivers/hmis_supplemental/app/controllers/hmis_supplemental/client_data_sets_controller.rb:45]: this was the bug

File: drivers/hmis_supplemental/app/controllers/hmis_supplemental/client_data_sets_controller.rb [hunk 2]
```diff
@@ -57,20 +58,6 @@ def load_authorized_data_set
       data_set_scope.find(params[:data_set_id])
     end
 
-    def client_groups
```
Comment (ttoomey) [inline drivers/hmis_supplemental/app/controllers/hmis_supplemental/client_data_sets_controller.rb:60]: unused code I forgot to remove 🤦

File: drivers/hmis_supplemental/spec/requests/client_data_set_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        expect(assigns(:groups).map { |h| h[:title] }).to include(
+          a_string_including(project1.name),
+          a_string_including(project2.name),
+        )
+        expect(assigns(:groups).size).to eq(2)
+      end
+    end
+  end
+
+  context 'without access' do
```
Comment (eanders) [inline drivers/hmis_supplemental/spec/requests/client_data_set_controller_spec.rb]: Isn't the check we need that when there is supplemental data on both the source data sources (`data_source` and `data_source_other`) but the user is only given access to one of them, that the `data_source_other` data isn't visible?  I might be missing something here, but I'd expect something like
```
expect(assigns(:groups).map { |h| h[:title] }).to_not include(
           a_string_including(project_other.name),
         )
```

Comment (ttoomey) [reply drivers/hmis_supplemental/spec/requests/client_data_set_controller_spec.rb]: The bug addressed in this PR was that the page was showing for all visible source clients (not just the one client that matching the data set's data source). So the bug wasn't really an issue related to data set visibility and it should be covered by the example above.

> Isn't the check we need that when there is supplemental data on both the source data sources (data_source and data_source_other)

So if you only have access to one data set, you would only see the enrollments for the source client in that data set? I'll adjust the test for that, probably a good idea

**Review by eanders — APPROVED** [review]
Thanks @ttoomey, looks good.

**Pull request merged**